libs/shared/sai-editor/src/sai-editor.service.spec.ts

```diff
@@ -144,5 +144,31 @@ describe('SAI Editor Service', () => {
       expect(getNameSpy).toHaveBeenCalledTimes(2);
       expect(generateCommentSpy).toHaveBeenCalledTimes(2);
     }));
+
+    it('should work correctly generating only one comment', fakeAsync(() => {
+      const mockRows: Partial<SmartScripts>[] = [
+        { entryorguid: 0, source_type: 0, id: 0, link: 1, event_type: 0 },
+        { entryorguid: 0, source_type: 0, id: 1, link: 0, event_type: 61 },
+      ];
+
+      service['_newRows'] = mockRows as SmartScripts[];
+
+      const updateDiffQuerySpy = spyOn<any>(service, 'updateDiffQuery');
+      const updateFullQuerySpy = spyOn<any>(service, 'updateFullQuery');
+      const refreshDatatableSpy = spyOn(service, 'refreshDatatable');
+      const isRowSelectedSpy = spyOn(service, 'isRowSelected').and.returnValue(true);
+      const getNameSpy = spyOn<any>(handlerService, 'getName');
+      const generateCommentSpy = spyOn<any>(saiCommentGeneratorService, 'generateComment');
+
+      service.generateComments();
+      tick(100);
+
+      expect(updateDiffQuerySpy).toHaveBeenCalledTimes(1);
+      expect(updateFullQuerySpy).toHaveBeenCalledTimes(1);
+      expect(refreshDatatableSpy).toHaveBeenCalledTimes(1);
+      expect(isRowSelectedSpy).toHaveBeenCalledTimes(1);
+      expect(getNameSpy).toHaveBeenCalledTimes(1);
+      expect(generateCommentSpy).toHaveBeenCalledTimes(1);
```

Comment:
shouldn't we also check on the actual comment that has been added?

Reply:
everything is mocked, so the comment is undefined here